### PR TITLE
Features/psp 2118

### PR DESCRIPTION
**Pull request by FuriousLlama**

### CODE REVIEW

File: backend/api/Areas/Leases/Controllers/SecurityDepositController.cs
Comment (devinleighsmith) [inline backend/api/Areas/Leases/Controllers/SecurityDepositController.cs:17]: nit: summary incorrect.

File: backend/api/Areas/Leases/Mapping/Lease/SecurityDepositMap.cs
Comment (devinleighsmith) [inline backend/api/Areas/Leases/Mapping/Lease/SecurityDepositMap.cs:20]: Do we have business approval on converting this many relationship to a 1-1? If so, do we need some method of sorting here?

File: backend/api/Areas/Leases/Mapping/Lease/SecurityDepositMap.cs
Comment (devinleighsmith) [reply backend/api/Areas/Leases/Mapping/Lease/SecurityDepositMap.cs:23]: Also, when selecting Organizations you are making sure to select only non-null values. Should there not be similar logic for the organization id? otherwise the organization you select may not be the same as the organization id that gets selected.

Comment (devinleighsmith) [reply backend/api/Areas/Leases/Mapping/Lease/SecurityDepositMap.cs:23]: ie: this case: [{organization: null, organizationId: 1}, {organization: {blah} organizationId: 2}]
I think this logic will get an organizationId of 1 but return organization blah

Comment (devinleighsmith) [reply backend/api/Areas/Leases/Mapping/Lease/SecurityDepositMap.cs:23]: Recommend writing a function that both can use to get the organization, and then for the id you can call getOrganization().id

File: backend/api/Areas/Leases/Mapping/Lease/SecurityDepositReturnMap.cs
Comment (devinleighsmith) [inline backend/api/Areas/Leases/Mapping/Lease/SecurityDepositReturnMap.cs:25]: Same issue here.

File: backend/api/Controllers/LookupController.cs
Comment (devinleighsmith) [inline backend/api/Controllers/LookupController.cs:93]: change to alphabetical? suppose that is fine.

File: backend/dal/IPimsRepository.cs
Comment (devinleighsmith) [inline backend/dal/IPimsRepository.cs:36]: I think you forgot to add the securityDepositReturnRepository?

File: backend/dal/Repositories/SecurityDepositRepository.cs
Comment (devinleighsmith) [inline backend/dal/Repositories/SecurityDepositRepository.cs:37]: Yes, please remove this.

File: backend/dal/Repositories/SecurityDepositRepository.cs
Comment (devinleighsmith) [inline backend/dal/Repositories/SecurityDepositRepository.cs:46]: ? Hopefully this can be removed as well.

File: backend/dal/Repositories/SecurityDepositRepository.cs
Comment (devinleighsmith) [inline backend/dal/Repositories/SecurityDepositRepository.cs:64]: Not a fan of leaving commented out code in committed source unless there is a good reason.

File: backend/dal/Repositories/SecurityDepositReturnRepository.cs
Comment (devinleighsmith) [inline backend/dal/Repositories/SecurityDepositReturnRepository.cs:66]: same comment here.

File: backend/dal/Services/SecurityDepositReturnService.cs
Comment (devinleighsmith) [inline backend/dal/Services/SecurityDepositReturnService.cs:27]: is this necessary? upstream of this in the controller you throw an exception if the lease id is mismatched right? maybe that logic should be here instead?

File: backend/dal/Services/SecurityDepositReturnService.cs
Comment (devinleighsmith) [inline backend/dal/Services/SecurityDepositReturnService.cs:61]: I need to confirm this with the business, but I've heard two different things in meetings now for delete:
1) if a user can edit a lease sub-entity they should be able to delete it.
2) a user should never be able to delete anything unless they have the delete permission.

Comment (devinleighsmith) [reply backend/dal/Services/SecurityDepositReturnService.cs:61]: sent an email and cc'd you on it. I've already updated terms with this assumption.

File: database/mssql/scripts/dbscripts/PSP_PIMS_S20_00/Test Data/03_DML_LEASE_TESTING.sql
Comment (devinleighsmith) [inline database/mssql/scripts/dbscripts/PSP_PIMS_S20_00/Test Data/03_DML_LEASE_TESTING.sql:1]: Thanks for adding this, I'm assuming that all of these scripts run successfully? (you may need to check the docker logs to confirm).

Comment (devinleighsmith) [reply database/mssql/scripts/dbscripts/PSP_PIMS_S20_00/Test Data/03_DML_LEASE_TESTING.sql:1]: In the future we probably want to do this in a separate PR.

File: frontend/package-lock.json
Comment (devinleighsmith) [inline frontend/package-lock.json:3]: hmm.

File: frontend/src/buttons.scss
Comment (devinleighsmith) [inline frontend/src/buttons.scss:87]: I feel like transparent may be safer choice here.

File: frontend/src/features/leases/detail/LeasePages/deposits/DepositsContainer.tsx
Comment (devinleighsmith) [inline frontend/src/features/leases/detail/LeasePages/deposits/DepositsContainer.tsx:118]: remove.

File: frontend/src/features/leases/detail/LeasePages/deposits/DepositsContainer.tsx
Comment (devinleighsmith) [inline frontend/src/features/leases/detail/LeasePages/deposits/DepositsContainer.tsx:132]: we can leave this in if you change it to a console.error

File: frontend/src/features/leases/detail/LeasePages/deposits/DepositsContainer.tsx
Comment (devinleighsmith) [inline frontend/src/features/leases/detail/LeasePages/deposits/DepositsContainer.tsx:88]: I'm kind of surprised you went with a static method here, wouldn't operating on the data in the instance make sense?

File: frontend/src/features/leases/detail/LeasePages/deposits/DepositsContainer.tsx
Comment (devinleighsmith) [reply frontend/src/features/leases/detail/LeasePages/deposits/DepositsContainer.tsx:132]: I think I'd be tempted to throw an exception in cases like this, otherwise the user will interact with the page and no visible changes will occur.

File: frontend/src/features/leases/detail/LeasePages/deposits/DepositsContainer.tsx
Comment (devinleighsmith) [inline frontend/src/features/leases/detail/LeasePages/deposits/DepositsContainer.tsx:167]: I'm not against adding logging like this to our backend and frontend but I don't think we want to do it via console logs. Will check if we have a tech debt item for logging on the FE (via a library).

File: frontend/src/features/leases/detail/LeasePages/deposits/DepositsContainer.tsx
Comment (devinleighsmith) [inline frontend/src/features/leases/detail/LeasePages/deposits/DepositsContainer.tsx:199]: can you just do onReturn={onReturnDeposit}
The additional wrapper function seems unecessary.

File: frontend/src/features/leases/detail/LeasePages/deposits/components/DepositsReceivedContainer/DepositsReceivedContainer.tsx
Comment (devinleighsmith) [inline frontend/src/features/leases/detail/LeasePages/deposits/components/DepositsReceivedContainer/DepositsReceivedContainer.tsx:34]: I don't think you need the {''}

File: frontend/src/features/leases/detail/LeasePages/deposits/components/DepositsReceivedContainer/DepositsReceivedContainer.tsx
Comment (devinleighsmith) [inline frontend/src/features/leases/detail/LeasePages/deposits/components/DepositsReceivedContainer/DepositsReceivedContainer.tsx:32]: SectiontHeader ? or just SectionHeader

File: frontend/src/features/leases/detail/LeasePages/deposits/components/DepositsReceivedContainer/columns.tsx
Comment (devinleighsmith) [inline frontend/src/features/leases/detail/LeasePages/deposits/components/DepositsReceivedContainer/columns.tsx:52]: may need to update delete claim based on outcome of email.

File: frontend/src/features/leases/detail/LeasePages/deposits/modal/receivedDepositModal/ReceivedDepositForm.test.tsx
Comment (devinleighsmith) [inline frontend/src/features/leases/detail/LeasePages/deposits/modal/receivedDepositModal/ReceivedDepositForm.test.tsx:21]: update name

File: frontend/src/features/leases/detail/LeasePages/deposits/modal/receivedDepositModal/ReceivedDepositForm.test.tsx
Comment (devinleighsmith) [inline frontend/src/features/leases/detail/LeasePages/deposits/modal/receivedDepositModal/ReceivedDepositForm.test.tsx:64]: Are these WIP?

File: frontend/src/features/leases/detail/LeasePages/deposits/modal/returnedDepositModal/ReturnDepositForm.tsx
Comment (devinleighsmith) [inline frontend/src/features/leases/detail/LeasePages/deposits/modal/returnedDepositModal/ReturnDepositForm.tsx:44]: I don't love the mix of bootstrap classnames and styled components.

File: backend/api/Areas/Leases/Controllers/SecurityDepositController.cs
Comment (FuriousLlama) [reply backend/api/Areas/Leases/Controllers/SecurityDepositController.cs:17]: updated.

File: backend/api/Areas/Leases/Mapping/Lease/SecurityDepositMap.cs
Comment (FuriousLlama) [reply backend/api/Areas/Leases/Mapping/Lease/SecurityDepositMap.cs:20]: This is one of those places where the db model does not match the requirements. It looks like the deposits can have multiple holders, however, the requirements imply that there is only one holder.

We will have to revisit the holder logic once we work on the contact manager story.

File: backend/api/Areas/Leases/Mapping/Lease/SecurityDepositMap.cs
Comment (FuriousLlama) [reply backend/api/Areas/Leases/Mapping/Lease/SecurityDepositMap.cs:23]: This is similar to the previous comment. I think we will have to chat about the db model to make sure it matches the requirements before making further changes to this.

File: backend/api/Areas/Leases/Mapping/Lease/SecurityDepositReturnMap.cs
Comment (FuriousLlama) [reply backend/api/Areas/Leases/Mapping/Lease/SecurityDepositReturnMap.cs:25]: yep, same as above

File: backend/api/Controllers/LookupController.cs
Comment (FuriousLlama) [reply backend/api/Controllers/LookupController.cs:93]: yeah, just an OCD thing since it look quite arbitrary

File: backend/dal/IPimsRepository.cs
Comment (FuriousLlama) [reply backend/dal/IPimsRepository.cs:36]: odd, it should be there

File: backend/dal/Repositories/SecurityDepositRepository.cs
Comment (FuriousLlama) [reply backend/dal/Repositories/SecurityDepositRepository.cs:37]: yeah, removed all the stubs for adding deposit holders since right now that functionality is not there.

File: backend/dal/Repositories/SecurityDepositRepository.cs
Comment (FuriousLlama) [reply backend/dal/Repositories/SecurityDepositRepository.cs:46]: Replied in the previous comment

File: backend/dal/Repositories/SecurityDepositReturnRepository.cs
Comment (FuriousLlama) [reply backend/dal/Repositories/SecurityDepositReturnRepository.cs:66]: Replied in the previous comment

File: backend/dal/Services/SecurityDepositReturnService.cs
Comment (FuriousLlama) [reply backend/dal/Services/SecurityDepositReturnService.cs:27]: fair enough, updated in the other places as well

File: database/mssql/scripts/dbscripts/PSP_PIMS_S20_00/Test Data/03_DML_LEASE_TESTING.sql
Comment (FuriousLlama) [reply database/mssql/scripts/dbscripts/PSP_PIMS_S20_00/Test Data/03_DML_LEASE_TESTING.sql:1]: Yeah, it worked fine. Thats a good point, I was hoping that the changes being SQL would make it obvious, but will keep them separate next time

File: frontend/src/buttons.scss
Comment (FuriousLlama) [reply frontend/src/buttons.scss:87]: Not 100% sure, but the secondary button has a white background on the mockups.

File: frontend/src/features/leases/detail/LeasePages/deposits/DepositsContainer.tsx
Comment (FuriousLlama) [reply frontend/src/features/leases/detail/LeasePages/deposits/DepositsContainer.tsx:118]: yeah, will remove them once the tests run sucessfuly

File: frontend/src/features/leases/detail/LeasePages/deposits/DepositsContainer.tsx
Comment (FuriousLlama) [reply frontend/src/features/leases/detail/LeasePages/deposits/DepositsContainer.tsx:132]: I think ideally we would have a toast for things like this since

File: frontend/src/features/leases/detail/LeasePages/deposits/DepositsContainer.tsx
Comment (FuriousLlama) [reply frontend/src/features/leases/detail/LeasePages/deposits/DepositsContainer.tsx:88]: thats right, I will change it to operate on the object directly

File: frontend/src/features/leases/detail/LeasePages/deposits/DepositsContainer.tsx
Comment (FuriousLlama) [reply frontend/src/features/leases/detail/LeasePages/deposits/DepositsContainer.tsx:167]: I agree, it is here only for debugging purposes while the tests are completed

File: frontend/src/features/leases/detail/LeasePages/deposits/DepositsContainer.tsx
Comment (FuriousLlama) [reply frontend/src/features/leases/detail/LeasePages/deposits/DepositsContainer.tsx:199]: true, will make it more straight forward

File: frontend/src/features/leases/detail/LeasePages/deposits/components/DepositsReceivedContainer/DepositsReceivedContainer.tsx
Comment (FuriousLlama) [reply frontend/src/features/leases/detail/LeasePages/deposits/components/DepositsReceivedContainer/DepositsReceivedContainer.tsx:32]: Good find, It looks like it was misspelled in several places

File: frontend/src/features/leases/detail/LeasePages/deposits/modal/receivedDepositModal/ReceivedDepositForm.test.tsx
Comment (FuriousLlama) [reply frontend/src/features/leases/detail/LeasePages/deposits/modal/receivedDepositModal/ReceivedDepositForm.test.tsx:64]: very much so

File: frontend/src/features/leases/detail/LeasePages/deposits/modal/returnedDepositModal/ReturnDepositForm.tsx
Comment (FuriousLlama) [reply frontend/src/features/leases/detail/LeasePages/deposits/modal/returnedDepositModal/ReturnDepositForm.tsx:44]: we should chat some more about this. Our styling is not consistent atm

File: backend/api/Areas/Leases/Mapping/Lease/SecurityDepositMap.cs
Comment (devinleighsmith) [reply backend/api/Areas/Leases/Mapping/Lease/SecurityDepositMap.cs:20]: Yes, lets bring this up during backlog refinement tomorrow.

File: backend/api/Areas/Leases/Mapping/Lease/SecurityDepositMap.cs
Comment (devinleighsmith) [reply backend/api/Areas/Leases/Mapping/Lease/SecurityDepositMap.cs:23]: not sure I agree on this one, regardless of whether mapping a many-to-one is correct, that mapping should return the same value consistently. You should not be returning the id from objectA while also returning objectB.

Comment (FuriousLlama) [reply backend/api/Areas/Leases/Mapping/Lease/SecurityDepositMap.cs:23]: I dont disagree with that. My point is that since those methods/members are related to the holder model is better if we wait until the model gets sorted out.

File: backend/dal/Services/SecurityDepositReturnService.cs
Comment (devinleighsmith) [reply backend/dal/Services/SecurityDepositReturnService.cs:61]: Confirmed this with the business. They honestly don't know if the ability to delete needs to be locked down. As a result, we are should control any delete functionality with the delete claim. For now, everyone who has the edit claim will also be given the delete claim, but in the future, that may be more locked down.

Comment (devinleighsmith) [reply backend/dal/Services/SecurityDepositReturnService.cs:61]: So yes, any delete functionality should be controlled by the delete claim not the edit claim.

Comment (devinleighsmith) [reply backend/dal/Services/SecurityDepositReturnService.cs:61]: @asanchezr

File: backend/api/Areas/Leases/Controllers/LeaseTermController.cs
Comment (devinleighsmith) [inline backend/api/Areas/Leases/Controllers/LeaseTermController.cs:14]: thanks for this

File: backend/dal/Services/SecurityDepositReturnService.cs
Comment (devinleighsmith) [reply backend/dal/Services/SecurityDepositReturnService.cs:59]: @FuriousLlama think this still needs to be updated to have delete controlled by delete claims.

File: frontend/src/features/leases/detail/LeasePages/deposits/DepositsContainer.tsx
Comment (devinleighsmith) [inline frontend/src/features/leases/detail/LeasePages/deposits/DepositsContainer.tsx]: oo, didn't catch this the first time around, glad to see it was removed.

**Review by devinleighsmith — COMMENTED** [review]
@FuriousLlama approved, assuming that the delete logic will be updated with the appropriate claim.

**Comment by devinleighsmith** [comment]
It does seem like this story is light on tests as well. If that is the case can you log tech debt for adding unit tests please?

**Review by devinleighsmith — APPROVED** [review]

**Pull request merged**
